server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java

```diff
@@ -180,22 +188,22 @@ public boolean mustAck(DiscoveryNode discoveryNode) {
                     public void onAllNodesAcked() {
                         ActiveShardsObserver.waitForActiveShards(
                             clusterService,
-                            new String[] { indexName },
+                            indexNames.toArray(String[]::new),
                             ActiveShardCount.DEFAULT,
                             request.timeout(),
                             allocationActionMultiListener.delay(listener)
-                                .map(shardsAcked -> new CreateIndexResponse(true, shardsAcked, indexName))
+                                .map(shardsAcked -> new CreateIndexResponse(true, shardsAcked, indexNames.get(0)))
                         );
                     }
 
                     @Override
                     public void onAckFailure(Exception e) {
-                        allocationActionMultiListener.delay(listener).onResponse(new CreateIndexResponse(false, false, indexName));
+                        allocationActionMultiListener.delay(listener).onResponse(new CreateIndexResponse(false, false, indexNames.get(0)));
                     }
 
                     @Override
                     public void onAckTimeout() {
-                        allocationActionMultiListener.delay(listener).onResponse(new CreateIndexResponse(false, false, indexName));
+                        allocationActionMultiListener.delay(listener).onResponse(new CreateIndexResponse(false, false, indexNames.get(0)));
```

Comment:
For these responses, what happens if the initial one is acknowledged (the regular `.ds-*` index), but the failure store index isn't acknowledged, do we ever have a case where we need to return that the create index response failed to acknowledge the `.fs-*` index? Or maybe we never run into that because we're always doing both at once?

Reply:
> Or maybe we never run into that because we're always doing both at once?

This is my understanding (though it may be flawed). Based on how I'm reading the code, we are applying the create operations all on one cluster state update. The metadata services apply the changes to the cluster state and the state update is submitted from the auto create operation. I don't think it's possible for one index to be created and the other not to be.

Now, I believe that it's possible for the shards observer to timeout, but shards being acked is not necessarily a required condition for noting success.